tests/behaviour/contracts/new/many.sol

```diff
@@ -0,0 +1,108 @@
+
+pragma solidity ^0.8.14;
+
+contract Uint256Contract {
+    uint public x;
+    constructor (uint x_) {
+        x = x_;
+    }
+    function getX() public view returns (uint256) {
+        return x;
+    }
+}
+
+contract Uint8Contract {
+    uint8 public x;
+    constructor (uint8 x_) {
+        x = x_;
+    }
+    function getX() public view returns (uint8) {
+        return x;
+    }
+}
+contract DynArrayContract {
+    uint[] public x ;
+    constructor (uint[] memory x_) {
+        x = x_;
+    }
+    function getX() public view returns (uint[] memory) {
+        return x;
+    }
+}
+contract StaticArrayContract {
+    uint8[3] public x;
+    constructor (uint8[3] memory x_) {
+        x = x_;
+    }
+    function getX() public view returns (uint8[3] memory) {
+        return x;
+    }
+}
+
+struct S {
+    uint8 m1;
+    uint8 m2;
+}
+
+contract StructContract {
+    S public x;
+    constructor (S memory x_){
+        x = x_;
+    }
+
+    function getX() public view returns (S memory) {
+        return x;
+    }
+}
+
+contract WARP {
+    Uint256Contract u256;
+    Uint8Contract u8;
+    DynArrayContract dac;
+    StaticArrayContract sac;
+    StructContract sc;
+
+    function createUint256Contract(uint x_) public {
+        u256 = new Uint256Contract(x_);
+    }
+
+    function createUint8Contract(uint8 x_) public {
+        u8 = new Uint8Contract(x_);
+    }
+
+    function createDynArrayContract(uint[] memory x_) public {
+        dac = new DynArrayContract(x_);
```

Comment:
Does the encode_to_felt work for calldata?

Reply:
It deals with any type. If not pass the right storage type a conversion is triggered.

Reply:
Great!